### PR TITLE
Port to 3.3 of various fixes around leadership preparation in agency.

### PR DESCRIPTION
**Pull request by neunhoef**

### CODE REVIEW

**Comment by matthewvon** [comment]
Line 235 concerns me.  Again I fear the value going negative from a clock shift.  This routine looks ready for steady clock (waitFor()).

**Comment by matthewvon** [comment]
And changing _lastSent[] to steady_clock makes the entire file steady_clock.  Might as well.

File: arangod/Agency/Node.h [hunk 1]
```diff
@@ -66,6 +66,7 @@ class StoreException : public std::exception {
 enum NODE_EXCEPTION { PATH_NOT_FOUND };
 
 typedef std::chrono::system_clock::time_point TimePoint;
```
Comment (matthewvon) [inline arangod/Agency/Node.h]: I think this can be removed and full conversion to SteadyTimePoint.

File: arangod/Scheduler/SchedulerFeature.cpp [hunk 2]
```diff
@@ -100,10 +100,9 @@ void SchedulerFeature::validateOptions(
 
   if (_nrMaximalThreads == 0) {
     _nrMaximalThreads = 4 * _nrServerThreads;
-  }
-
-  if (_nrMaximalThreads < 64) {
-    _nrMaximalThreads = 64;
+    if (_nrMaximalThreads < 64) {
```
Comment (matthewvon) [inline arangod/Scheduler/SchedulerFeature.cpp]: P.S.  I now understand this.  Thank you for prior comments.

**Comment by kvahed** [comment]
cherry picked changes from #4147

**Pull request merged**